src/transaction/declare.rs

```diff
@@ -136,29 +121,19 @@ impl Declare {
             skip_fee_transfer: false,
         };
 
-        verify_version(
-            &internal_declare.version,
-            internal_declare.max_fee,
-            &internal_declare.nonce,
-            &internal_declare.signature,
-        )?;
-
-        Ok(internal_declare)
-    }
-
-    pub fn get_calldata(&self) -> Vec<Felt252> {
-        let bytes = Felt252::from_bytes_be(&self.class_hash);
-        Vec::from([bytes])
+        Ok(declare)
     }
 
     /// Executes a call to the cairo-vm using the accounts_validation.cairo contract to validate
     /// the contract that is being declared. Then it returns the transaction execution info of the run.
-    pub fn apply<S: State + StateReader>(
+    fn validate<S: State + StateReader>(
         &self,
         state: &mut S,
         block_context: &BlockContext,
     ) -> Result<TransactionExecutionInfo, TransactionError> {
+        // verify the version and check the nonce
         verify_version(&self.version, self.max_fee, &self.nonce, &self.signature)?;
+        handle_nonce(&self.nonce, &self.version, &self.sender_address, state)?;
```

Comment:
I should move this handle nonce to the fn execute functions, 
I think it is more clear